Wikipedia/Code/WMFRelatedSectionController.m

```diff
@@ -121,57 +99,74 @@ - (NSAttributedString*)headerText {
-    [tableView registerNib:[WMFArticlePlaceholderTableViewCell wmf_classNib] forCellReuseIdentifier:[WMFArticlePlaceholderTableViewCell identifier]];
+- (nullable UINib*)placeholderCellNib {
+    return [WMFArticlePlaceholderTableViewCell wmf_classNib];
 }
 
-- (UITableViewCell*)dequeueCellForTableView:(UITableView*)tableView atIndexPath:(NSIndexPath*)indexPath {
-    if ([self hasResults]) {
-        return [WMFArticlePreviewTableViewCell cellForTableView:tableView];
-    } else {
-        return [WMFArticlePlaceholderTableViewCell cellForTableView:tableView];
-    }
+- (void)configureCell:(WMFArticlePreviewTableViewCell*)cell withItem:(MWKSearchResult*)item atIndexPath:(NSIndexPath*)indexPath {
+    cell.titleText       = item.displayTitle;
+    cell.descriptionText = item.wikidataDescription;
+    cell.snippetText     = item.extract;
+    [cell setImageURL:item.thumbnailURL];
+    [cell setSaveableTitle:[self titleForItemAtIndexPath:indexPath] savedPageList:self.savedPageList];
+    [cell wmf_layoutIfNeededIfOperatingSystemVersionLessThan9_0_0];
+    cell.saveButtonController.analyticsSource = self;
 }
 
-- (void)configureCell:(UITableViewCell*)cell withObject:(id)object inTableView:(UITableView*)tableView atIndexPath:(NSIndexPath*)indexPath {
-    if ([cell isKindOfClass:[WMFArticlePreviewTableViewCell class]] && [object isKindOfClass:[MWKSearchResult class]]) {
-        WMFArticlePreviewTableViewCell* previewCell = (id)cell;
-        MWKSearchResult* result                     = object;
-        previewCell.titleText       = result.displayTitle;
-        previewCell.descriptionText = result.wikidataDescription;
-        previewCell.snippetText     = result.extract;
-        [previewCell setImageURL:result.thumbnailURL];
-        [previewCell setSaveableTitle:[self titleForItemAtIndex:indexPath.row] savedPageList:self.savedPageList];
-        [previewCell wmf_layoutIfNeededIfOperatingSystemVersionLessThan9_0_0];
-        previewCell.saveButtonController.analyticsSource = self;
-    }
+- (CGFloat)estimatedRowHeight {
+    return [WMFArticlePreviewTableViewCell estimatedRowHeight];
+}
+
+- (NSString*)analyticsName {
+    return @"Related";
 }
 
-- (BOOL)shouldSelectItemAtIndex:(NSUInteger)index {
-    return [self hasResults];
+- (AnyPromise*)fetchData {
+    @weakify(self);
+    return [self.relatedTitleDataSource fetch]
+           .then(^(WMFRelatedSearchResults* results){
+        @strongify(self);
+        self.searchResults = results;
+        return [self.searchResults.results wmf_safeSubarrayWithRange:NSMakeRange(0, WMFRelatedSectionMaxResults)];
+    })
+           .catch(^(NSError* error){
+        @strongify(self);
+        self.searchResults = nil;
+        return error;
+    });
+}
+
+#pragma mark - WMFHeaderMenuProviding
+
+- (UIActionSheet*)menuActionSheet {
+    UIActionSheet* sheet = [[UIActionSheet alloc] bk_initWithTitle:nil];
+    [sheet bk_addButtonWithTitle:MWLocalizedString(@"home-hide-suggestion-prompt", nil) handler:^{
```

Comment:
@coreyfloyd @nirzar this should be a destructive button (`bk_setDestructiveButtonWithTitle`)